src/pilgrim/utils/config_manager.py

```
@@ -0,0 +1,107 @@
+        self.auto_open_new_diary = None
+        self.config_dir = DirectoryManager.get_config_directory()
+        self.__data = None
+
+    def read_config(self):
+        if os.path.exists(f"{DirectoryManager.get_config_directory()}/config.toml"):
+            try:
+                with open(f"{DirectoryManager.get_config_directory()}/config.toml", "rb") as f:
+                    data = tomli.load(f)
+
+            except tomli.TOMLDecodeError as e:
+                raise ValueError(f"Invalid TOML configuration: {e}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)